packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -493,54 +493,31 @@ class RangeMaintainingScrollPhysics extends ScrollPhysics {
     required bool isScrolling,
     required double velocity,
   }) {
-    bool maintainOverscroll = true;
     bool enforceBoundary = true;
     if (velocity != 0.0) {
       // Don't try to adjust an animating position, the jumping around
       // would be distracting.
-      maintainOverscroll = false;
       enforceBoundary = false;
     }
-    if ((oldPosition.minScrollExtent == newPosition.minScrollExtent) &&
-        (oldPosition.maxScrollExtent == newPosition.maxScrollExtent)) {
-      // If the extents haven't changed then ignore overscroll.
-      maintainOverscroll = false;
-    }
-    if (oldPosition.pixels != newPosition.pixels) {
-      // If the position has been changed already, then it might have
-      // been adjusted to expect new overscroll, so don't try to
-      // maintain the relative overscroll.
-      maintainOverscroll = false;
-      if (oldPosition.minScrollExtent.isFinite && oldPosition.maxScrollExtent.isFinite &&
-          newPosition.minScrollExtent.isFinite && newPosition.maxScrollExtent.isFinite) {
-        // In addition, if the position changed then we only enforce
-        // the new boundary if the previous boundary was not entirely
-        // finite. A common case where the position changes while one
-        // of the extents is infinite is a lazily-loaded list. (If the
-        // boundaries were finite, and the position changed, then we
-        // assume it was intentional.)
-        enforceBoundary = false;
-      }
-    }
     if ((oldPosition.pixels < oldPosition.minScrollExtent) ||
         (oldPosition.pixels > oldPosition.maxScrollExtent)) {
       // If the old position was out of range, then we should
       // not try to keep the new position in range.
       enforceBoundary = false;
     }
-    if (maintainOverscroll) {
-      // Force the new position to be no more out of range
-      // than it was before, if it was overscrolled.
-      if (oldPosition.pixels < oldPosition.minScrollExtent) {
-        final double oldDelta = oldPosition.minScrollExtent - oldPosition.pixels;
-        return newPosition.minScrollExtent - oldDelta;
-      }
-      if (oldPosition.pixels > oldPosition.maxScrollExtent) {
-        final double oldDelta = oldPosition.pixels - oldPosition.maxScrollExtent;
-        return newPosition.maxScrollExtent + oldDelta;
-      }
+    if (oldPosition.pixels != newPosition.pixels &&
+        oldPosition.minScrollExtent.isFinite && oldPosition.maxScrollExtent.isFinite &&
+        newPosition.minScrollExtent.isFinite && newPosition.maxScrollExtent.isFinite) {
+      // Enforce the boundary if the position has been changed already and
+      // both the previous and new positions are entirely finite.
+      //
+      // A common case where the position changes while one
+      // of the extents is infinite is a lazily-loaded list. (If the
+      // boundaries were finite, and the position changed, then we
+      // assume it was intentional.)
+      enforceBoundary = false;
```

Comment:
The comment from above looked wrong, I tried to adapt it, but it was hard as there's no test for this, as far as I can tell. It would be nice if @Hixie checked I changed it correctly.